index.js

```diff
@@ -1,37 +1,62 @@
 var uuid = require('node-uuid');
 var CRLF = '\r\n';
 
+function compact(array) {
+  return array.filter(function(element) {
+    return typeof element !== 'undefined' && element !== null;
```

Comment:
To also filter empty strings I think that you could simply use
```javascript
return array.filter(function(element) { return !!element; }) 
```

At which point you could simply put this below, where you use `compact`:

```javascript
var multipartBody = [
    preamble ? CRLF + preamble : undefined,
    encapsulations.join(''),
    closeDelimiter,
    epilogue ? CRLF + epilogue : undefined,
 ].filter(function(element) { return !!element; });
```